src/apps/users/tests/test_password.py

```diff
@@ -0,0 +1,177 @@
+            **self.create_request_user.dict()
+        )
+
+        # User get token
+        await self.client.get_token(
+            url=self.get_token_url,
+            user_login_request=login_request_user,
+        )
+
+        # Password update
+        response = await self.client.put(
+            self.password_update_url, data=self.password_update_request.dict()
+        )
+
+        updated_user: User = await UsersCRUD().get_by_email(
+            login_request_user.email
+        )
+
+        public_user = PublicUser(**updated_user.dict())
+
+        expected_result: Response[PublicUser] = Response(result=public_user)
+
+        # User get token with new password
+        login_request_user: UserLoginRequest = UserLoginRequest(
+            email=self.create_request_user.dict()["email"],
+            password=self.password_update_request.dict()["password"],
+        )
+
+        internal_response = await self.client.get_token(
+            url=self.get_token_url,
+            user_login_request=login_request_user,
+        )
+
+        assert response.status_code == status.HTTP_200_OK
+        assert response.json() == expected_result.dict(by_alias=True)
+        assert response.status_code == status.HTTP_200_OK
+        assert internal_response.status_code == status.HTTP_200_OK
+
+    @transaction.rollback
+    @patch("apps.users.services.core.MailingService.send")
+    @patch("apps.users.services.core.PasswordRecoveryCache.set")
+    @patch("apps.users.services.core.PasswordRecoveryCache.delete_all_entries")
+    async def test_password_recovery(
+        self,
+        cache_delete_all_entries_mock,
+        cache_set_mock,
+        mailing_send_mock,
+    ):
+        # Creating new user
+        internal_response: Response[PublicUser] = await self.client.post(
+            self.user_create_url, data=self.create_request_user.dict()
+        )
+
+        expected_result = internal_response.json()
+
+        # Password recovery
+        password_recovery_request: PasswordRecoveryRequest = (
+            PasswordRecoveryRequest(
+                email=self.create_request_user.dict()["email"]
+            )
+        )
+
+        response = await self.client.post(
+            url=self.password_recovery_url,
+            data=password_recovery_request.dict(),
+        )
+
+        assert (
+            cache_delete_all_entries_mock
+            is PasswordRecoveryCache.delete_all_entries
+        )
+        assert cache_set_mock is PasswordRecoveryCache.set
+        assert mailing_send_mock is MailingService.send
+        assert cache_delete_all_entries_mock.call_count == 1
+        assert cache_set_mock.call_count == 1
+        assert mailing_send_mock.call_count == 1
+        assert response.status_code == status.HTTP_200_OK
+        assert response.json() == expected_result
+
+    @transaction.rollback
+    @patch("apps.users.services.core.PasswordRecoveryCache.delete_all_entries")
```

Comment:
Why have you used patch? it is already mocked, not needed to patch it

Reply:
We should use mock objects not only to ignore some functionality.
Check the whole test. Here I use this patch in order to check how many calls we did to that mocked object.
`cache_delete_all_entries_mock`

Reply:
I know that we can get results from mocks, but it gives what we expect. If it is changed in lower level, in that case your test will pass successfully, but result will be another. Moreover, you have to mock it everywhere, it makes hard refactoring. you can resolve if you wish

Reply:
Because of your base testing class, I can not easily use the pytest-mocker package's fixtures. That's why I have to use this built-in patch method.

> I know that we can get results from mocks, but it gives what we expect.
No, it is not. As you can see I count the number of calls. It could be 0 if the function wasn't called.

> Moreover, you have to mock it everywhere, it makes hard refactoring. you can resolve if you wish
No, it is mocked on more abstract level. We have to use this mock only in places we have to check the calls number or passed arguments. And I don't understand why it should make the refactoring more complicated :)